lib/trento/application/integration/checks/adapter/gen.ex

```diff
@@ -10,4 +10,10 @@ defmodule Trento.Integration.Checks.Gen do
               selected_checks :: [String.t()]
             ) ::
               :ok | {:error, any}
+
+  @callback get_runner_ready_content(runner_url :: String.t()) ::
+              {:ok, term} | {:error, any}
+
+  @callback get_catalog_content(runner_url :: String.t()) ::
+              {:ok, term} | {:error, any}
```

Comment:
`@callback get_catalog(runner_url :: String.t()) :: {:ok, map} | {:error, :not_ready} | {:error, any}`

Reply:
I will change this to `@callback get_catalog :: {:ok, term} | {:error, :not_ready} | {:error, any}` as the catalog conversion will happen in the main `Checks` module

Reply:
nitpick: `{:ok, any}`. It is  actually equivalent to term, but term is more an erlang thing and usually elixir specs use `any`

Reply:
here you could also use `{:ok, map}`